servicetalk-encoding-netty/src/main/java/io/servicetalk/encoding/netty/NettyCompressionStreamingSerializer.java

```diff
@@ -80,8 +83,21 @@ public void onNext(@Nullable final Buffer next) {
                 }
 
                 try { // onNext will produce AT-MOST N items (as received)
-                    channel.writeInbound(extractByteBufOrCreate(next));
-                    next.skipBytes(next.readableBytes());
+                    final Buffer bufferToDecode;
+                    if (aggregate != null && aggregate.readableBytes() != 0) {
+                        aggregateNext(next);
+                        bufferToDecode = aggregate;
```

Comment:
Consider chaning `aggregateNext` signature to return `Buffer`. Then we can do `bufferToDecode = aggregateNext(next);`